backend/tools/event_sender/src/main.rs

```diff
@@ -2,24 +2,39 @@ use std::{
     fs::File,
     io::{BufRead, BufReader},
     time::Duration,
+    time::Instant,
 };
 
 use eiffelvis_gen::{
     event_set::{Event, EventSet, Link},
     generator::EventGenerator,
 };
+
 use lapin::{options::*, BasicProperties, Connection, ConnectionProperties};
 
 use clap::Parser;
 use rand::{thread_rng, Rng};
 
+const EVENT_TYPES: [&str; 23] = [
```

Comment:
Can we put everything related to the EventSet in its own dedicated file?
As I mentioned in #14 making a zst type that implements From<> would be nice.

```rust
pub struct EIffelVocabulary;

const EVENT_TYPES: [&str; 23] = [
    "ActC", "ActF", "ActS", "ActT", "AnnP", "ArtC", "ArtP", "ArtR", "TCC", "TCF", "TCS", "TCT",
    "TERCC", "TSF", "TSS", "CD", "CLM", "ED", "FCD", "ID", "IV", "SCC", "SCS",
];

impl From<EIffelVocabulary> for EventSet {
    fn from(_: EIffelVocabulary) -> Self {
        let mut builder = EventSet::build();

        for eventtype in EVENT_TYPES {
            // Initialize and create the event variable
            let mut event = Event::new(eventtype.to_string(), "1.0.0");

            // Create the random number generate for the links
            let _randomrange = rand::thread_rng().gen_range(1..3);

            // Loop and add the links to the event
            for linknumber in 0.._randomrange {
                event = event.with_link(format!("Link{linknumber}"));
            }

            builder = builder.add_event(event);
        }

        builder
            .add_link(Link::new("Link0", true))
            .add_link(Link::new("Link1", true))
            .add_event(
                Event::new("Event", "1.0.0")
                    .with_link("Link0")
                    .with_link("Link1"),
            )
            .build()
            .expect("This should work")
    }
}

```

then in the main.rs file:

```rust
let gen = EventGenerator::new(
        cli.seed.unwrap_or_else(|| thread_rng().gen::<usize>()),
        4,
        8,
        EIffelVocabulary.into()
    );
```

Reply:
We will try to do this this week, depending on the schedule!

Reply:
Added the changes suggested.